_posts/2015-11-13-svg-sprites-angular.md

```diff
@@ -0,0 +1,167 @@
+    </symbol>
+    <symbol id="icon-social-twitter" viewBox="0 0 512 512">
+        <path d="m256 0c-141 0-256 115-256 256s115 256 256 256 256-115 256-256-115-256-256-256zm104 210v6c0 67-50 144-143 144-29 0-55-9-77-23 3 1 7 1 12 1 23 0 45-8 62-22-22 0-41-15-47-35 3 1 6 1 10 1s9-1 13-2c-23-4-41-25-41-49v-1c7 4 15 6 23 7-13-10-22-25-22-42 0-10 2-18 7-26 24 31 62 51 104 53-1-4-2-8-2-12 0-27 23-50 51-50 14 0 27 6 36 16 12-2 23-7 33-12-4 11-12 21-23 28 11-2 20-4 29-8-6 10-15 19-25 26z"/>
+    </symbol>
+</svg>
+
+```
+
+And that is already our svg sprite. Now we just have to know how to reference it. Cool thing about svg elements is, that we can reference them by id (for with css sprites, you need to specify 'coordinates' of specific icon as background position), so to get a facebook icon, we can just do this
+
+
+```
+<svg class="svg-icon">
+    <use xlink:href="#icon-social-facebook"></use>
+</svg>
+
+```
+
+We can apply some css properties to it (to svg-icon class) and style it the way we want and that would be it. With that little change, you can improve the way you handle your assets and make it much simpler to use in different context.
+
+
+## Gotchas - is not always roses
+
+- Internet explorer (of course) will not load svgs if you have them in external file (not embedded in template itself)
+- Safari will not load your icons if they are referenced before they are defined (embedded after called)
+- < IE9 is not supporting svgs
+- Using `<base>` changes behavior in firefox (more below)
+
+Currently, we are embedding svg sprites in our template, just after opening `<body>` tag for good cross browser support. If you want to keep you resources out of your template, you can use a great shim written by Jonathan Neal [svg4everybody] (https://github.com/jonathantneal/svg4everybody).
+
+And if you need to support older browsers (IE8 etc.) you can use image as fallback.
+
+## Examples
+
+[Here] (https://github.com/SinisaG/svg-blog-demo) are a few examples that shows some of the browser glitches.
+
+You can run the examples using [5minutefork] (http://5minfork.com/). More in readme....
+
+## Base tag
+
+If you are like us and use [AngularJs] (https://angularjs.org/) everything written above will apply if you are using urls with hashbang (#!). But if you are using pretty urls with `<base>` tag or you use this tag for any other reason, then be aware, that firefox is interpreting this differently then other browsers, so you will need to reference svgs with an absolute url. For angular or similar environments we suggest using a directive that keeps urls up to date:
+
+
+```javascript
```

Comment:
Isn't that something that we could / should add to https://github.com/Onefootball/onefootball-angular-components ?